src/System.Reflection.TypeExtensions/src/System/Reflection/TypeExtensions.CoreCLR.cs

```diff
@@ -270,6 +270,41 @@ public static MethodInfo GetRemoveMethod(EventInfo eventInfo, bool nonPublic)
         }
     }
 
+    public static class MemberInfoExtensions
+    {
+        public static bool TryGetMetadataToken(this MemberInfo member, out int metadataToken)
+        {
+            Requires.NotNull(member, "member");
+
+            int token;
+            try
+            {
+                token = member.MetadataToken;
+            }
+            catch (InvalidOperationException)
+            {
+                // Thrown for unbaked ref-emit members/types. 
+                // Surface as failed TryGet.
+                metadataToken = 0;
```

Comment:
I guess it is these special cases that made you stick with the TryGet pattern instead of using Nullable<int>? I don't have a strong opinion either way, just curious.